shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/connection/ConnectionManager.java

```diff
@@ -168,6 +171,73 @@ public void rollback() throws SQLException {
         }
     }
     
+    /**
+     * Rollback to savepoint.
+     *
+     * @param savepoint savepoint
+     * @throws SQLException SQL exception
+     */
+    public void rollback(final Savepoint savepoint) throws SQLException {
+        if (!connectionTransaction.isInTransaction()) {
+            throw new SQLException("Now, not in transaction");
```

Comment:
> this log is a little bit simple, why not print out some more detail info of Savepoint for a better debugging ?

This is an exception thrown and returned directly to the client. It may be better to be simpler.

Reply:
It is better to rewrite the error message, the current one is too unnormalized.

Reply:
> It is better to rewrite the error message, the current one is too unnormalized.

@terrymanu Different DB have different behaviors. MySQL can execute successfully and PG fails. How about use the error message of PG directly?
```error: savepoint can only be used in transaction blocks```